src/containerapp/azext_containerapp/tests/latest/utils.py

```diff
@@ -69,4 +72,162 @@ def create_and_verify_containerapp_up(
+        app = test_cls.cmd(f"containerapp show -g {resource_group} -n {app_name}", checks=[
+            JMESPathCheck('properties.configuration.registries[0].server', registry_server),
+            JMESPathCheck('properties.configuration.registries[0].username', registry_user),
+            JMESPathCheck('properties.provisioningState', 'Succeeded'),
+        ]).get_output_in_json()
+        test_cls.assertEqual(app["properties"]["template"]["containers"][0]["image"].split(":")[0], image_name)
+
+        # Re-construct the 'az containerapp create' command with registry username and password
+        create_cmd = 'containerapp create -g {} -n {} --environment {} --registry-username {} --registry-server {} --registry-password {}'.format(
+            resource_group, app_name, env_name, registry_user, registry_server, registry_pass)
+        if source_path:
+            create_cmd += f" --source \"{source_path}\""
+        if image:
+            create_cmd += f" --image {image}"
+            image_name = registry_server + "/" + _reformat_image(image)
+        if ingress:
+            create_cmd += f" --ingress {ingress}"
+        if target_port:
+            create_cmd += f" --target-port {target_port}"
+
+        # Re-execute the 'az containerapp create' command
+        test_cls.cmd(create_cmd)
+
+        # Verify successful execution
+        app = test_cls.cmd(f"containerapp show -g {resource_group} -n {app_name}", checks=[
+            JMESPathCheck('properties.configuration.registries[0].server', registry_server),
+            JMESPathCheck('properties.configuration.registries[0].username', registry_user),
+            JMESPathCheck('properties.provisioningState', 'Succeeded'),
+        ]).get_output_in_json()
+        test_cls.assertEqual(app["properties"]["template"]["containers"][0]["image"].split(":")[0], image_name)
+        old_image = app["properties"]["template"]["containers"][0]["image"]
+
+        # Construct the 'az containerapp update' command
+        update_cmd = 'containerapp update -g {} -n {}'.format(
+            resource_group, app_name)
+        if source_path:
+            update_cmd += f" --source \"{source_path}\""
+        if image:
+            update_cmd += f" --image {image}"
+
+        # Execute the 'az containerapp update' command
+        test_cls.cmd(update_cmd)
+
+        # Verify successful execution
+        app = test_cls.cmd(f"containerapp show -g {resource_group} -n {app_name}", checks=[
+            JMESPathCheck('properties.configuration.registries[0].server', registry_server),
+            JMESPathCheck('properties.configuration.registries[0].username', registry_user),
+            JMESPathCheck('properties.provisioningState', 'Succeeded'),
+        ]).get_output_in_json()
+        test_cls.assertEqual(app["properties"]["template"]["containers"][0]["image"].split(":")[0], image_name)
+        test_cls.assertNotEqual(app["properties"]["template"]["containers"][0]["image"], old_image)
```

Comment:
add container app delete and env delete commands to make sure all resources deleted

Reply:
Updated.